### PR TITLE
Feat/2822 write files in json

### PR DESCRIPTION
**Pull request by adipaul1981**

### CODE REVIEW

File: src/main/scala/io/kf/etl/ETLMain.scala [hunk 1]
```diff
@@ -5,8 +5,13 @@ import io.kf.etl.processors.download.DownloadProcessor
 import io.kf.etl.processors.featurecentric.FeatureCentricProcessor
 import io.kf.etl.processors.index.IndexProcessor
 import io.kf.etl.processors.participantcommon.ParticipantCommonProcessor
+import io.kf.etl.processors.tojson.JsonOutputProcessor
+
+import scala.util.Try
 
 object ETLMain extends App {
+  val Array(saveJsonFiles) = args
+  val saveJsonToS3: Boolean = Try(saveJsonFiles.toBoolean).getOrElse(false)
```
Comment (jecos) [inline src/main/scala/io/kf/etl/ETLMain.scala:14]: instead having a new parameter, could we relied on the same config that conatinas the filepath?

File: src/test/scala/io/kf/etl/processors/participantcommon/transform/step/WriteJsonSinkTest.scala [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+  "process" should "merge phenotypes and participant" in {
+    val dataset = Seq(
+      ParticipantCentric_ES(kf_id = Some("p1"), gender = Some("male")),
+      ParticipantCentric_ES(kf_id = Some("p2"), gender = Some("female"), alias_group = Some("ag1")),
+      ParticipantCentric_ES(kf_id = Some("p2"), gender = Some("male"), alias_group = Some("ag2"), is_proband = Some(true))
+    ).toDS()
+
+    WriteJsonSink.exportDataSetToJsonFile("./src/test/json_output", "file_centric_study1_release1")(dataset)
+
+    val directory = new Directory(new File("./src/test/json_output/file_centric_study1_release1"))
```
Comment (jecos) [inline src/test/scala/io/kf/etl/processors/participantcommon/transform/step/WriteJsonSinkTest.scala:22]: use getClass().getRessource(...) instead

File: src/test/scala/io/kf/etl/processors/participantcommon/transform/step/WriteJsonSinkTest.scala [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+import io.kf.etl.processors.test.util.WithSparkSession
+import org.scalatest.{FlatSpec, Matchers}
+
+import java.io.File
+import scala.reflect.io.Directory
+
+class WriteJsonSinkTest extends FlatSpec with Matchers with WithSparkSession {
+import spark.implicits._
+
+  "process" should "merge phenotypes and participant" in {
```
Comment (jecos) [inline src/test/scala/io/kf/etl/processors/participantcommon/transform/step/WriteJsonSinkTest.scala:13]: Not sure the test is relevant

**Review by jecos — CHANGES_REQUESTED** [review]

**Review by botekchristophe — APPROVED** [review]

File: src/main/scala/io/kf/etl/processors/participantcommon/transform/step/WriteJsonSink.scala [hunk 4]
```diff
@@ -8,8 +8,7 @@ object WriteJsonSink {
   def apply[T](processorName: String)(step: String)(input: Dataset[T])(implicit config: Config, spark: SparkSession): Dataset[T] = {
```
Comment (jecos) [inline src/main/scala/io/kf/etl/processors/participantcommon/transform/step/WriteJsonSink.scala:8]: I would verify if this class is already used. If not, I would rather implement export to jsin directly into JsonOutputProcessor and delete this class

**Review by jecos — CHANGES_REQUESTED** [review]

File: src/main/scala/io/kf/etl/ETLMain.scala [hunk 5]
```diff
@@ -1,17 +1,24 @@
...
+import scala.util.Try
 
 object ETLMain extends App {
 
   DefaultContext.withContext { context =>
     import context.implicits._
 
     import scala.concurrent.ExecutionContext.Implicits._
+
+    val saveJsonToS3 = Try(config.getBoolean(SAVE_JSON_FILES)).getOrElse(false)
```
Comment (jecos) [inline src/main/scala/io/kf/etl/ETLMain.scala:20]: Maybe we could have only one env var :
```
     val saveJsonToS3 = Try(config.getString(JSON_OUTPUT_FILES)).map(s=> s!= null && s.nonEmpty).getOrElse(false)  
```

**Review by jecos — CHANGES_REQUESTED** [review]

**Pull request merged**